From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Branson <andrew.branson@jolla.com>
Date: Thu, 6 Aug 2026 13:49:35 +0200
Subject: [sailfishos][media] Keep AsyncDBus GTK-only

Qt builds can enable both D-Bus and PipeWire, but AsyncDBus is only
built and exported by the GTK widget backend. Keep the camera portal
probe under the GTK guard so Qt falls back to the existing V4L2 path.
---
 .../systemservices/video_engine/video_capture_factory.cc    | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dom/media/systemservices/video_engine/video_capture_factory.cc b/dom/media/systemservices/video_engine/video_capture_factory.cc
index 374320dc3ebbe..e6bd8864021bd 100644
--- a/dom/media/systemservices/video_engine/video_capture_factory.cc
+++ b/dom/media/systemservices/video_engine/video_capture_factory.cc
@@ -17,7 +17,8 @@
 #  include "video_engine/placeholder_device_info.h"
 #endif
 
-#if defined(WEBRTC_USE_PIPEWIRE) && defined(MOZ_ENABLE_DBUS)
+#if defined(WEBRTC_USE_PIPEWIRE) && defined(MOZ_ENABLE_DBUS) && \
+    defined(MOZ_WIDGET_GTK)
 #  include "mozilla/widget/AsyncDBus.h"
 #endif
 
@@ -147,7 +148,8 @@ auto VideoCaptureFactory::InitCameraBackend()
 
 auto VideoCaptureFactory::HasCameraDevice()
     -> RefPtr<VideoCaptureFactory::HasCameraDevicePromise> {
-#if defined(WEBRTC_USE_PIPEWIRE) && defined(MOZ_ENABLE_DBUS)
+#if defined(WEBRTC_USE_PIPEWIRE) && defined(MOZ_ENABLE_DBUS) && \
+    defined(MOZ_WIDGET_GTK)
   if (mVideoCaptureOptions && mVideoCaptureOptions->allow_pipewire()) {
     return widget::CreateDBusProxyForBus(
                G_BUS_TYPE_SESSION, G_DBUS_PROXY_FLAGS_NONE,
